plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCOrderStore.kt

```diff
@@ -857,7 +857,8 @@ class WCOrderStore @Inject constructor(
 
     private suspend fun revertOrderStatus(payload: RemoteOrderPayload.Updating): OnOrderChanged {
         updateOrderStatusLocally(payload.order.orderId, payload.order.localSiteId, payload.order.status)
-        return OnOrderChanged().also { it.error = payload.error }
+        return OnOrderChanged(causeOfChange = WCOrderAction.UPDATE_ORDER_STATUS)
+            .also { it.error = payload.error }
```

Comment:
Oh ok @atorresveiga I see. That's because we update here `OnChanged#error`, which is not part of `data class OnOrderChanged`. That's quite a tricky API. I've introduced it :( but it's still better than before IMO. I think we're prisoners of `OnChanged` super class here.

My suggestion would be to use following constructor:
```suggestion
        return OnOrderChanged(orderError = payload.error)
```

and remove change from line 575-576. I think that this way the code is more readable.

Either way I'd appreciate unit tests as you agreed, so we'll make sure that no one falls into this trap again.